src/main/kotlin/app/revanced/patches/youtube/layout/returnyoutubedislike/patch/ReturnYouTubeDislikePatch.kt

```diff
@@ -78,39 +80,69 @@ class ReturnYouTubeDislikePatch : BytecodePatch(
-            }.result!!.scanResult.patternScanResult!!.endIndex
+        }.result?.also { result ->
+            if (!TextComponentAtomicReferenceFingerprint.resolve(context, result.method, result.classDef))
+                throw TextComponentAtomicReferenceFingerprint.toErrorResult()
+        }?.let { textComponentContextFingerprintResult ->
+            // Get the instruction indices of the conversion context and the span reference.
+            val conversionContextIndex = textComponentContextFingerprintResult
+                .scanResult.patternScanResult!!.startIndex
+            val spanReferenceEndIndex = TextComponentAtomicReferenceFingerprint.result!!
+                .scanResult.patternScanResult!!.endIndex
+
+            textComponentContextFingerprintResult.mutableMethod.apply {
+                // Get the registers of the conversion context and the span reference.
+                val conversionContextRegister =
+                    (instruction(conversionContextIndex) as TwoRegisterInstruction).registerA
+                val atomicReferenceRegister =
+                    (instruction(spanReferenceEndIndex) as TwoRegisterInstruction).registerB
+
+                // Insert the call to onComponentCreated after the span reference is created to change the dislike count.
+                val insertIndex = spanReferenceEndIndex + 1
+                addInstruction(
+                    insertIndex,
+                    "invoke-static {v$conversionContextRegister, v$atomicReferenceRegister}, " +
+                            "$INTEGRATIONS_PATCH_CLASS_DESCRIPTOR->" +
+                            "onComponentCreated(Ljava/lang/Object;Ljava/util/concurrent/atomic/AtomicReference;)V"
+                )
+            }
+        } ?: return TextComponentContextFingerprint.toErrorResult()
 
-            result.mutableMethod.apply {
-                val insertIndex = result.scanResult.patternScanResult!!.endIndex
+        // endregion
 
-                val atomicReferenceInstruction = (instruction(insertIndex - 1) as FiveRegisterInstruction)
-                val conversionContextParam = atomicReferenceInstruction.registerC
-                val textRefParam = (instruction(moveTextRefParamInstructionIndex) as TwoRegisterInstruction).registerA
+        // region Hook all cases, except when a video is disliked
 
-                // Overwritten after injected code, which is why it can be used.
-                val clobberRegister = atomicReferenceInstruction.registerD
+        TextComponentSpecFingerprint.result?.let {
+            with(it.mutableMethod) {
+                val returnSpanIndex = it.scanResult.patternScanResult!!.endIndex
+                val spannedStringRegister = (instruction(returnSpanIndex) as OneRegisterInstruction).registerA
+                // Store the context in any free register, which is any register except the one already used by the return value
+                val contextTempRegister = spannedStringRegister.xor(1)
```

Comment:
I may note this might malfunction if `spannedStringRegister` is anything but 0 or 1.